internal/checker/checker.go

```diff
@@ -14497,6 +14497,34 @@ func (c *Checker) resolveExternalModule(location *ast.Node, moduleReference stri
 						tsExtension,
 					)
 				}
+			} else if c.compilerOptions.RewriteRelativeImportExtensions.IsTrue() &&
+				location.Flags&ast.NodeFlagsAmbient == 0 &&
+				!tspath.IsDeclarationFileName(moduleReference) &&
+				!ast.IsLiteralImportTypeNode(location) &&
+				!ast.IsPartOfTypeOnlyImportOrExportDeclaration(location) {
+				shouldRewrite := c.shouldRewriteModuleSpecifier(moduleReference)
+				if !resolvedModule.ResolvedUsingTsExtension && shouldRewrite {
+					relativeToSourceFile := tspath.GetRelativePathFromDirectory(
```

Comment:
It looks like the diffs didn't fully disappear because `GetRelativePathFromDirectory` has slightly different semantics from the original `getRelativePathFromFile`. Go ahead and port `getRelativePathFromFile` (you can locate the original function in `_submodule/TypeScript`)